addons/crm/models/crm_lead.py

```diff
@@ -418,7 +419,10 @@ def _compute_lang_id(self):
         else:
             lang_id_by_code = {}
         for lead in self.filtered('partner_id'):
-            lead.lang_id = lang_id_by_code.get(lead.partner_id.lang, False)
+            if lead.partner_id.lang:
+                lead.lang_id = lang_id_by_code.get(lead.partner_id.lang, False)
+            else:
+                lead.lang_id = lead.lang_id
```

Comment:
as partner_id is set anyway, you could directly do:
`lead.lang_id = lang_id_by_code.get(lead.partner_id.lang, lead.lang_id)`